static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js

```diff
@@ -134,13 +133,13 @@ const shouldShowEpic = (test: EpicABTest): boolean => {
 
     const tagsMatch = doTagsMatch(test);
 
-    return (
-        shouldShowReaderRevenue(test.showToContributorsAndSupporters) &&
-        onCompatiblePage &&
-        inCompatibleLocation &&
-        test.locationCheck(storedGeolocation) &&
-        tagsMatch
-    );
+    return !shouldHideReaderRevenue && test.showToContributorsAndSupporters
```

Comment:
I'm not sure this logic is right.

The `&&` has [higher precedence](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence) than `?:` so this would mean that if `test.showToContributorsAndSupporters` is true it would bypass all the `onCompatiblePage`, `inCompatibleLocation` etc checks, which is not what we want

Something like this might be better:
```javascript
const isCompatibleUser = test.showToContributorsAndSupporters ? userIsSupporter() : !userIsSupporter();

return 
  !shouldHideReaderRevenue &&
  onCompatiblePage &&
  inCompatibleLocation &&
  isCompatibleUser &&
  test.locationCheck(storedGeolocation) &&
  tagsMatch 
```